### PR TITLE
Implement Authd tests code improvements.

### PR DESCRIPTION
**Pull request by palaciosjeremias**
|Related issue|
|---|
|#2184|

## Description
This PR implements all the pending suggestions from https://github.com/wazuh/wazuh-qa/pull/2171.
Basically:
- It fixes all the code style observations.
- Relocates **Authd.py** methods related to **Wazuh-db** into **wazuh-db.py**
- Relocates **Authd.py** fixtures into the **Authd** contest .
- Refactor **validate_authd_response()** logic to be more readable.
- Removes a deprecated README file with the test group definition.

## Tests

- [x] Proven that tests **pass** when they have to pass.
- [x] Proven that tests **fail** when they have to fail.

- [X] Python codebase satisfies PEP-8 style style guide. `pycodestyle --max-line-length=120 --show-source --show-pep8 file.py`.
- [X] Python codebase is documented following the Google Style for Python docstrings.
- [X] The test is documented in following the DocGenerator schema V2.0

### CODE REVIEW

**Comment by palaciosjeremias** [comment]
Succesful execution of Authd tests on Jenkins:
https://ci.wazuh.info/job/Test_integration/15575/

File: tests/integration/test_authd/conftest.py [hunk 1]
```diff
@@ -149,3 +149,42 @@ def override_authd_force_conf(format_configuration):
...
+    try:
+        keys_file = open(CLIENT_KEYS_PATH, 'w')
+    except IOError as exception:
+        raise exception
+
+    clean_agents_from_db()
+
+    for agent in agents:
+        id = agent['id'] if 'id' in agent else '001'
+        name = agent['name'] if 'name' in agent else f'TestAgent{id}'
```
Comment (juliamagan) [inline tests/integration/test_authd/conftest.py:167]:
```suggestion
        name = agent['name'] if 'name' in agent else f"TestAgent{id}"
```

File: tests/integration/test_authd/conftest.py [hunk 2]
```diff
@@ -149,3 +149,42 @@ def override_authd_force_conf(format_configuration):
...
+            disconnection_time = time_now
+        if 'registration_time' in agent and 'delta' in agent['registration_time']:
+            registration_time = time_now + agent['registration_time']['delta']
+        elif 'registration_time' in agent and 'value' in agent['registration_time']:
+            registration_time = agent['registration_time']['value']
+        else:
+            registration_time = time_now
+
+        # Write agent in client.keys
+        keys_file.write(f'{id} {name} {ip} {key}\n')
```
Comment (juliamagan) [inline tests/integration/test_authd/conftest.py:185]:
```suggestion
        keys_file.write(f"{id} {name} {ip} {key}\n")
```

**Review by juliamagan — CHANGES_REQUESTED** [review]

**Comment by juliamagan** [comment]
# 10/11/2021

## Package
| Version | Revision | Link|
|---|---|---|
4.3.0|40301|https://packages-dev.wazuh.com/warehouse/test/4.3/rpm/var/wazuh-manager-4.3.0-2192.authd.improve.x86_64.rpm|

## Testing

### tests/integration/test_authd

|  OS    | Local | Jenkins     | Notes
|---      |---    |---    |---  |
| PS1   |  ||
| PS2   |  |  |
| PS3   |  |  |   |

### tests/integration/test_wazuh_db

|  OS    | Local | Jenkins     | Notes
|---      |---    |---    |---  |
| PS1   |  | |
| PS2   |  |  |
| PS3   |   |  |

* * * 

- :green_circle:: All pass
- :yellow_circle:: Some warnings
- :red_circle:: Some errors/fails
- :large_blue_circle:: In progress

**Pull request merged**